synth_machine/synth_parser.py

```diff
@@ -0,0 +1,323 @@
+import xmltodict
+from copy import deepcopy
+from partial_json_parser import loads, OBJ
+import re
+
+try:
+    from enum import StrEnum
+except ImportError:
+    # For python versions <3.11, using aenum for backward compatibility
+    from aenum import StrEnum
+
+
+class ParserOptions(StrEnum):  # type: ignore
+    STR = "str"
+    CODE = "code"
+    JSON = "json"
+    XML = "xml"
+
+
+class SynthParser:
+    def __init__(self):
+        self.parse = {
+            "json": self.json_parse,
+            "xml": self.xml_parse,
+            "code": self.code_parse,
+        }
+
+    def json_parse(self, raw_value: str) -> dict | list:
+        return loads(str(raw_value), OBJ)
+
+    def xml_parse(self, raw_value: str) -> dict:
+        open_tag_pattern = r"<[^/][^>]*>"
```

Comment:
Why are we doing extra processing on the xml? Surely we should just support what the library does and if your format is wrong we present the error from the library?

Reply:
Partial xml parsing to work with streaming responses